src/Utilities/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -285,6 +285,38 @@ public static DisposeMethodKind GetDisposeMethodKind(this IMethodSymbol method,
             return DisposeMethodKind.None;
         }
 
+        /// <summary>
+        /// Checks if the given method implements IDeserializationCallback.OnDeserialization()
+        /// </summary>
+        public static bool IsOnDeserializationImplementation(this IMethodSymbol method, Compilation compilation)
+        {
+            INamedTypeSymbol IDeserializationCallback = WellKnownTypes.IDeserializationCallback(compilation);
```

Comment:
Avoid computing well known types in helper methods, instead cache them in analyzer's compilation start action, and pass the type symbol instead of compilation to helper methods

Reply:
Ah, I think you should just delete this overload `public static bool IsOnDeserializationImplementation(this IMethodSymbol method, Compilation compilation)` and force all analyzers to call the below overload `public static bool IsOnDeserializationImplementation(this IMethodSymbol method, INamedTypeSymbol iDeserializationCallback)`